src/browser/rectangle.ts

```diff
@@ -600,9 +589,11 @@ function propMoveThroughGraph (
     let movedRef = rects[refVertex];
 
     if (movedRef.hasIdenticalBounds( cachedBounds)) {
+        // This is the leader, move it to the proposed bounds
```

Comment:
Should we check if the new bounds violate size constraints or is that taken care of? @MichaelMCoates

Reply:
Seems to be good in this regard

Reply:
My size constraint violation stuff was only on API calls.